documentation/source/user_section/tutorials/shape-analysis/normalizing-morphometrics-compressions/generating-necessary-inputs.rst

```diff
@@ -0,0 +1,84 @@
+.. _generating-necessary-inputs:
+
+Generate the necessary input files
+#############################################################
+
+The ``sct_compute_compression`` command uses the shape metric of the spinal cord at the level of compression and computes a ratio with non-compressed levels above and below the compression site.
+
+
+To use ``sct_compute_compression``, we need the 3 following input files:
+
+
+1. Spinal cord segmentation
+2. Vertebral labels
+3. Compression labels (``t2_compressed_label-compression.nii.gz``)
+
+You can also get these files by downloading :sct_tutorial_data:`data_normalizing-morphometrics-compression.zip`.
+
+1. Spinal cord segmentation
+----------------------------
+.. code:: sh
+
+   sct_deepseg_sc -i t2_compressed.nii.gz -c t2 -qc ~/qc_singleSubj
+
+:Input arguments:
+   - ``-i`` : Input image.
+   - ``-c`` : Contrast of the input image.
+   - ``-qc`` : Directory for Quality Control reporting.
+
+
+:Output files/folders:
+   - ``t2_compressed_seg.nii.gz`` : Spinal cord segmentation.
+
+
+2. Label the spinal cord vertebral levels
+------------------------------------------
+.. code:: sh
+
+   sct_label_vertebrae -i t2_compressed.nii.gz -s t2_compressed_seg.nii.gz -c t2 -qc ~/qc_singleSubj
+
+:Input arguments:
+   - ``-i`` : The input image file.
+   - ``-s`` : Spinal cord segmentation.
+   - ``-c`` : Contrast of the input image.
+   - ``-qc`` : Directory for Quality Control reporting.
+
+:Output files/folders:
+   - ``t2_compressed_seg_labeled.nii.gz`` : Spinal cord vertebral labels.
+
+3. Generate spinal cord compression labels
+-------------------------------------------
+
+1. Open the image in FSLeyes. If the image does not have sufficient axial and sagittal resolution, open two individual images (axial image and sagittal image) in FSLeyes.
```

Comment:
This is a very basic question, but: What does this comment mean?

I ask because, in the example file (`t2_compression.nii.gz`), I'm assuming we do have sufficient resolution, so this advice doesn't apply here.

Because I don't have a concrete example, I don't know what the alternate case ("not sufficient resolution") even looks like. I also don't know what it means to open two individual images in FSLeyes. Does the user need to have two separate images? Does the cursor sync up when they're both opened at once What if they don't have two images?

Reply:
Excellent question! Thanks!

> I ask because, in the example file (t2_compression.nii.gz), I'm assuming we do have sufficient resolution, so this advice doesn't apply here.

Correct, `t2_compressed.nii.gz` has 0.8x0.8x0.8 mm resolution. The resolution is so-called isotropic (i.e., equivalent in all planes). It allows both 1) the computation of morphometric measures from the axial plane and 2) the identification of vertebral levels and compression level(s) in the sagittal plane. This is why we need only a single image here.

> I don't know what the alternate case ("not sufficient resolution") even looks like. I also don't know what it means to open two individual images in FSLeyes.

I admit that the term "not sufficient resolution" is not very specific here. It was supposed to mean that the image is, for example, axially acquired with a resolution of 0.5x0.5x4 mm. In such an example, 0.5x0.5 is great for the computation of morphometric measures, but 4mm slice thickness (5x higher compared to the 0.8mm image) is not ideal for vertebral levels and compression level(s). Therefore, we would open two individual images (for example, one sagittal and one axial) instead.

> Does the user need to have two separate images?

Yes. 

> Does the cursor sync up when they're both opened at once? What if they don't have two images?

Yes, FSLeyes allows you to open multiple images in a single session. The cursor is synchronized across them.

---

The approach with two images also requires their coregistration using `identity 1` during processing (see example [here](https://github.com/sct-pipeline/dcm-metric-normalization/blob/d985ac0ca24d6253bfe717310f4c6a3529243b60/scripts/process_data_dcm-zurich.sh#L157-L164)) and bringing sagittal disc labels to axial image (see [here](https://github.com/sct-pipeline/dcm-metric-normalization/issues/9)). Since this is a kind of "non-standard" or "advanced" processing, which we do not cover in other tutorials, we can maybe totally remove the sentence about two images to not confuse users:

```
1. Open the image in FSLeyes.
```

Reply:
Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4162/commits/b2406ecfbbff0891c2e421948bcab42c0e2ee408